### PR TITLE
chore: add go test, golangci-lint in github action

### PR DESCRIPTION
**Pull request by haunt98**
Add github action for:
- go test
- gotip build
- golangci-lint

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
```
Comment (cuonglm) [inline .github/workflows/go.yaml:13]: We have to make sure the code is still compile with gotip, see `ci/run` script for details.

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Go
+
+on: push
```
Comment (cuonglm) [inline .github/workflows/go.yaml:3]: Also, please make it run on every PR or on master branch

File: .github/workflows/go.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+      - run: go version
+      - run: go test -race -v ./...
+  build:
```
Comment (cuonglm) [inline .github/workflows/go.yaml:17]: I don't think we need this step? Any reason for adding this?

Comment (haunt98) [reply .github/workflows/go.yaml:17]: I think it good to test for bulding on many OS

File: .github/workflows/go.yaml (hunk 2 above)
Comment (haunt98) [reply .github/workflows/go.yaml]: I change it to run on master and P

File: .github/workflows/go.yaml (hunk 1 above)
Comment (haunt98) [reply .github/workflows/go.yaml:16]: I try to use gotip by `gotip download` but it kinda sl

File: .github/workflows/go.yaml (hunk 3 above)
Comment (cuonglm) [reply .github/workflows/go.yaml:19]: It does not make sense, though. If we want to test multiple OSes, we want 32-bit vs 64-bit system, though.

Comment (haunt98) [reply .github/workflows/go.yaml]: I've removed it

**Review by cuonglm — APPROVED** [review]

**Comment by odeke-em** [comment]
Thank you for working on this change @haunt98, and thank you for the review and merge @cuonglm!

**Pull request merged**